### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by andreaskuster**
The code overall looks good to me.

There is one class that I currently still get a headache and which is why I keep this PR as a draft for now (I hope you can shine some ligth into this @KarlSchu ).

It is the [`Console.cpp`](https://github.com/SolarEnergyRacers/solar_car_control_system/blob/master/lib/Console/Console.cpp) class.

(1) Why are `invalidChar` and `stripUnicode` not members of the class? https://github.com/SolarEnergyRacers/solar_car_control_system/blob/master/lib/Console/Console.cpp#L16

(2) I do not really understand why and how this works, however, it looks very weird to me from a concept perspective:
 - Why are we using the external console instance in the implementation (.cpp) section of the class? (https://github.com/SolarEnergyRacers/solar_car_control_system/blob/master/lib/Console/Console.cpp#L14)

- Why are these not class-functions? Why are they just globally? https://github.com/SolarEnergyRacers/solar_car_control_system/blob/master/lib/Console/Console.cpp#L16

- This is also not a class function here (i.e. not ClassName::ClassFunctionImplementation()..). Why? https://github.com/SolarEnergyRacers/solar_car_control_system/blob/master/lib/Console/Console.cpp#L23

### CODE REVIEW

**Review by KarlSchu — APPROVED** [review]

**Review by KarlSchu — APPROVED** [review]

File: lib/ADC/ADC.cpp [hunk 1]
```diff
@@ -82,7 +82,6 @@ int16_t ADC::read(ADC::Pin port) {
   value = ADC::adss[idx].readADC(pin);
   xSemaphoreGive(i2cBus.mutex);
 
-  // console << fmt::sprintf("index: 0x%x, pin: 0x%x => value=%d\n", idx, pin, value);
```
Comment (KarlSchu) [inline lib/ADC/ADC.cpp:85]: need for debug: console until now missing the debuglevel function :(

File: lib/CarControl/CarControl.cpp [hunk 2]
```diff
@@ -143,10 +143,6 @@ bool CarControl::read_paddles() {
   }
 
   if (valueDisplayLast != valueDisplay) {
-    // debug_printf("Dec (v0):  %5d --> %3d | Acc (v1): %5d --> %3d | ACCEL-DISPLAY: %3d"
-    //              " ==> set POT0 =%3d (dec(%5d-%5d)), POT1 =%3d (acc(%5d-%5d))\n",
-    //              valueDec, valueDecNorm, valueAcc, valueAccNorm, valueDisplay, valueDecPot, ads_min_dec, ads_max_dec, valueAccPot,
-    //              ads_min_acc, ads_max_acc);
```
Comment (KarlSchu) [inline lib/CarControl/CarControl.cpp:149]: need for debug: console until now missing the debuglevel function :(

File: lib/CarControl/CarControl.cpp [hunk 3]
```diff
@@ -155,9 +151,6 @@ bool CarControl::read_paddles() {
 }
 
 void CarControl::_set_dec_acc_values(int valueDecPot, int valueAccPot, int16_t valueDec, int16_t valueAcc, int valueDisplay) {
-  // debug_printf("Dec (v0):  %5d  | Acc (v1): %5d  | ACCEL-DISPLAY: %3d ==> set POT0 =%3d (dec(%5d-%5d)), POT1 =%3d (acc(%5d-%5d))\n",
-  //              valueDec, valueAcc, valueDisplay, valueDecPot, ads_min_dec, ads_max_dec, valueAccPot, ads_min_acc, ads_max_acc);
-
```
Comment (KarlSchu) [inline lib/CarControl/CarControl.cpp:160]: need for debug: console until now missing the debuglevel function :(

File: lib/DAC/DAC.cpp [hunk 4]
```diff
@@ -103,13 +101,11 @@ void DAC::set_pot(uint8_t val, pot_chan channel) {
...
     }
   }
-  // if (val > DAC_MAX)
-  //   val = DAC_MAX;
   // setup command
   uint8_t command = get_cmd(channel);
   uint8_t oldValue = get_pot(channel);
   if (oldValue != val) {
-    //#SAVETY#: Reset constant mode on deceleration paddel touched
+    // SAFETY: Reset constant mode on deceleration paddle touched
```
Comment (KarlSchu) [inline lib/DAC/DAC.cpp:108]: if you change #SAVETY# to SAVETY you mustdo it everywhere. My idea was to use a unique pattern for find all operations

**Review by KarlSchu — CHANGES_REQUESTED** [review]

**Comment by andreaskuster** [comment]
@KarlSchu All mentioned issues should be resolved

**Review by KarlSchu — APPROVED** [review]

**Pull request merged**